controllers/argocd/notifications/secret.go

```diff
@@ -0,0 +1,76 @@
+package notifications
+
+import (
+	"github.com/argoproj-labs/argocd-operator/pkg/cluster"
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	"github.com/argoproj-labs/argocd-operator/pkg/workloads"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (nr *NotificationsReconciler) reconcileSecret() error {
+
+	nr.Logger.Info("reconciling secrets")
+
+	secretRequest := workloads.SecretRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resourceName,
```

Comment:
this secret needs to be called `argocd-notifications-secret`, and can't follow the generic resource naming system
Please create a new constant in `constants.go` for `NotificationsSecretName` and use that here instead